src/Cassandra.IntegrationTests/Core/ConnectionSimulacronTests.cs

```diff
@@ -0,0 +1,422 @@
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Cassandra.Connections;
+using Cassandra.IntegrationTests.TestClusterManagement.Simulacron;
+using Cassandra.Tasks;
+using Cassandra.Tests;
+
+using Castle.Core;
+
+using NUnit.Framework;
+
+namespace Cassandra.IntegrationTests.Core
+{
+    public class ConnectionSimulacronTests : SimulacronTest
+    {
+        public ConnectionSimulacronTests() : base(false, new SimulacronOptions { Nodes = "3" }, false)
+        {
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        [Test]
+        public async Task Should_ThrowOperationTimedOut_When_ServerAppliesTcpBackPressure(bool streamMode)
+        {
+            SetupNewSession(b =>
+                b.WithPoolingOptions(
+                     new PoolingOptions()
+                         .SetCoreConnectionsPerHost(HostDistance.Local, 1)
+                         .SetMaxConnectionsPerHost(HostDistance.Local, 1))
+                 .WithSocketOptions(new SocketOptions()
+                                    .SetReadTimeoutMillis(1000)
+                                    .SetStreamMode(streamMode)
+                                    .SetDefunctReadTimeoutThreshold(int.MaxValue)));
+
+            var maxRequestsPerConnection = Session.Cluster.Configuration
+                                                  .GetOrCreatePoolingOptions(Session.Cluster.Metadata.ControlConnection.ProtocolVersion)
+                                                  .GetMaxRequestsPerConnection();
+            var tenKbBuffer = new byte[10240];
+
+            await TestCluster.PauseReadsAsync().ConfigureAwait(false);
+
+            // send number of requests = max pending
+            var requests =
+                Enumerable.Repeat(0, maxRequestsPerConnection * Session.Cluster.AllHosts().Count)
+                          .Select(i => Session.ExecuteAsync(new SimpleStatement("INSERT INTO table1 (id) VALUES (?)", tenKbBuffer))).ToList();
+
+            var taskAll = Task.WhenAll(requests);
+            try
+            {
+                await (await Task.WhenAny(taskAll, Task.Delay(60000)).ConfigureAwait(false)).ConfigureAwait(false);
+                Assert.Fail("Should time out.");
+            }
+            catch (NoHostAvailableException)
+            {
+                // ignored
+            }
+
+            requests = requests.Where(t => t.IsFaulted).ToList();
+
+            Assert.Greater(requests.Count, 1);
+
+            Assert.IsTrue(requests.All(
+                t => ((NoHostAvailableException)t.Exception.InnerException)
+                        .Errors.Any(e => e.Value is OperationTimedOutException)));
+        }
+
+        [TestCase(false)]
+        [TestCase(true)]
+        [Test]
+        public async Task Should_RetryOnNextNodes_When_ANodeIsPaused(bool streamMode)
+        {
+            var pausedNode = TestCluster.GetNode(2);
+
+            SetupNewSession(b =>
```

Comment:
Hmm I like using the default LBP (and default everything in general) in integration tests whenever possible because it might help identify regressions in the future. In some cases using an ordered LBP dramatically simplifies a test or makes it more reliable and in those cases +1 on going that route but this test case is easy to implement in a reliable way without changing those things.